pkg/virtctl/create/vm.go

```diff
@@ -0,0 +1,362 @@
+  {{ProgramName}} create vm
+    
+  # Create a manifest for a running VirtualMachine with a specified name
+  {{ProgramName}} create vm --name=my-vm --running=true
+
+  # Create a manifest for a VirtualMachine with a specified name and RunStrategy Always
+  {{ProgramName}} create vm --name=my-vm --run-strategy=Always
+  
+  # Create a manifest for a VirtualMachine with a specified VirtualMachineClusterInstancetype
+  {{ProgramName}} create vm --instancetype=my-instancetype
+
+  # Create a manifest for a VirtualMachine with a specified VirtualMachineInstancetype (namespaced)
+  {{ProgramName}} create vm --instancetype=virtualmachineinstancetype/my-instancetype
+  
+  # Create a manifest for a VirtualMachine with a specified VirtualMachineClusterPreference
+  {{ProgramName}} create vm --preference=my-preference
+
+  # Create a manifest for a VirtualMachine with a specified VirtualMachinePreference (namespaced)
+  {{ProgramName}} create vm --preference=virtualmachinepreference/my-preference
+
+  # Create a manifest for a VirtualMachine with a specified VirtualMachineCluster{Instancetype,Preference} and cloned DataSource
+  {{ProgramName}} create vm --instancetype=my-instancetype --preference=my-preference --datasource=my-ds
+
+  # Create a manifest for a VirtualMachine with a specified IVirtualMachineCluster{Instancetype,Preference} and cloned DataSource in namespace
+  {{ProgramName}} create vm --instancetype=my-instancetype --preference=my-preference --datasource=my-ns/my-ds
+
+  # Create a manifest for a VirtualMachine with a specified VirtualMachineCluster{Instancetype,Preference} and directly used PVC
+  {{ProgramName}} create vm --instancetype=my-instancetype --preference=my-preference --volume=my-pvc`
+}
+
+func splitPrefixedName(prefixedName string) (prefix string, name string, err error) {
+	split := strings.Split(prefixedName, "/")
+
+	switch l := len(split); l {
+	case 1:
+		name = split[0]
+	case 2:
+		prefix = split[0]
+		name = split[1]
+	default:
+		return "", "", fmt.Errorf("invalid count %d of slashes in prefix/name", l)
+	}
+
+	if name == "" {
+		return "", "", errors.New("name cannot be empty")
+	}
+
+	return
+}
+
+func newVM(c *createVM) *v1.VirtualMachine {
+	runStrategy := v1.VirtualMachineRunStrategy(c.runStrategy)
+	return &v1.VirtualMachine{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       v1.VirtualMachineGroupVersionKind.Kind,
+			APIVersion: v1.VirtualMachineGroupVersionKind.GroupVersion().String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: c.name,
+		},
+		Spec: v1.VirtualMachineSpec{
+			RunStrategy: &runStrategy,
+			Template: &v1.VirtualMachineInstanceTemplateSpec{
+				Spec: v1.VirtualMachineInstanceSpec{
+					TerminationGracePeriodSeconds: &c.terminationGracePeriod,
+				},
+			},
+		},
+	}
+}
+
+func withRunStrategy(c *createVM, vm *v1.VirtualMachine) error {
+	if _, ok := runStrategies[c.runStrategy]; !ok {
+		return fmt.Errorf("unknown RunStrategy %s", c.runStrategy)
```

Comment:
nit - maybe dump out the accepted `runStrategies` as part of the error for the user?